src/response_decoder.ts

```diff
@@ -84,10 +88,17 @@ function decodeResponse(
     }
     return { size, key, version, metadataInfo } as RawMetadataUpdateResponse
   }
+
+  if (key === CreditResponse.key) {
+    const responseCode = dataResponse.readUInt16()
+    const subscriptionId = dataResponse.readUInt8()
+    return { size, key, version, responseCode, subscriptionId } as RawCreditResponse
```

Comment:
I suggest using a different way:
```suggestion
   const ret: RawCreditResponse = { size, key, version, responseCode, subscriptionId }
    return  ret
```
with the `as` if we change a field or make some refactoring, the compiler doesn't raise an error